### PR TITLE
Move w2d to dynamic locale imports

### PR DESCRIPTION
**Pull request by bearfriend**
Uses the new `localize-dynamic-mixin` for true dynamic imports. Also enables OSLO.

~Dependent upon https://github.com/BrightspaceUI/core/pull/1146~

### CODE REVIEW

File: components/d2l-work-to-do/lang/en.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ export const val = {
 	Checklist : "Checklist", // Meta-data descriptor that informs which type of activity is being displayed on a line item
 	comeBackNoFutureActivities : "Come back later to see if you have work to do.", // 'Empty state' - When there are no activities in full page view
 	Content : "Content", // Meta-data descriptor that informs which type of activity is being displayed on a line item
-	Course : "Course", // Meta-data descriptor that informs which type of activity is being displayed on a line item
+	Course : "Course^^", // Meta-data descriptor that informs which type of activity is being displayed on a line item
```
Comment (dlockhart) [inline components/d2l-work-to-do/lang/en.js:9]: Was this by accident?

File: .eslintrc.json [hunk 2]
```diff
@@ -4,9 +4,10 @@
     "es6": true,
     "jest": true
   },
-  "parser": "babel-eslint",
+  "parser": "@babel/eslint-parser",
   "parserOptions": {
-      "sourceType": "module"
+      "sourceType": "module",
+			"requireConfigFile": false
```
Comment (dlockhart) [inline .eslintrc.json:10]: Nit: indentation

File: package.json [hunk 3]
```diff
@@ -23,6 +23,7 @@
   "license": "Apache-2.0",
   "devDependencies": {
     "@babel/core": "^7.8.3",
+    "@babel/eslint-parser": "^7.12.17",
```
Comment (dlockhart) [inline package.json:26]: Can the `babel-eslint` dependency (below) go away?

File: components/d2l-work-to-do/lang/en.js (hunk 1 above)
Comment (bearfriend) [reply components/d2l-work-to-do/lang/en.js:9]: 🤦

File: components/d2l-work-to-do/mixins/d2l-work-to-do-localization-mixin.js [hunk 4]
```diff
@@ -1,86 +1,11 @@
...
-
-		translations = await import('../lang/en.js');
-		return await getLocalizeOverrideResources(
-			'en',
-			translations.val,
-			resolveOverridesFunc
-		);
+	static get localizeConfig() {
+		return {
+			importFunc: async lang => (await import(`../lang/${lang}.js`)).val,
```
Comment (dlockhart) [inline components/d2l-work-to-do/mixins/d2l-work-to-do-localization-mixin.js:7]: 🎉

**Review by dlockhart — APPROVED** [review]

**Pull request merged**